nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ModifyCompression.java

```diff
@@ -0,0 +1,453 @@
+                            final int xzCompressionLevel = context.getProperty(COMPRESSION_LEVEL).asInteger();
+                            compressionOut = new XZOutputStream(bufferedOut, new LZMA2Options(xzCompressionLevel));
+                            mimeTypeRef.set("application/x-xz");
+                            break;
+                        case COMPRESSION_FORMAT_SNAPPY:
+                            compressionOut = new SnappyOutputStream(bufferedOut);
+                            mimeTypeRef.set("application/x-snappy");
+                            break;
+                        case COMPRESSION_FORMAT_SNAPPY_HADOOP:
+                            compressionOut = new SnappyHadoopCompatibleOutputStream(bufferedOut);
+                            mimeTypeRef.set("application/x-snappy-hadoop");
+                            break;
+                        case COMPRESSION_FORMAT_SNAPPY_FRAMED:
+                            compressionOut = new SnappyFramedOutputStream(bufferedOut);
+                            mimeTypeRef.set("application/x-snappy-framed");
+                            break;
+                        case COMPRESSION_FORMAT_LZ4_FRAMED:
+                            mimeTypeRef.set("application/x-lz4-framed");
+                            compressionOut = new CompressorStreamFactory().createCompressorOutputStream(compressionFormat.toLowerCase(), bufferedOut);
+                            break;
+                        case COMPRESSION_FORMAT_ZSTD:
+                            final int zstdcompressionLevel = context.getProperty(COMPRESSION_LEVEL).asInteger() * 2;
+                            compressionOut = new ZstdCompressorOutputStream(bufferedOut, zstdcompressionLevel);
+                            mimeTypeRef.set("application/zstd");
+                            break;
+                        case COMPRESSION_FORMAT_BROTLI:
+                            Brotli4jLoader.ensureAvailability();
+                            compressionLevel = context.getProperty(COMPRESSION_LEVEL).asInteger();
+                            Encoder.Parameters params = new Encoder.Parameters().setQuality(compressionLevel);
+                            compressionOut = new BrotliOutputStream(bufferedOut, params);
+                            mimeTypeRef.set("application/x-brotli");
+                            break;
+                        case COMPRESSION_FORMAT_BZIP2:
+                            mimeTypeRef.set("application/x-bzip2");
+                            compressionOut = new CompressorStreamFactory().createCompressorOutputStream(compressionFormat.toLowerCase(), bufferedOut);
+                            break;
+                        case COMPRESSION_FORMAT_NONE:
+                        default:
+                            compressionOut = bufferedOut;
+                            break;
+                    }
+
+                } catch (final Exception e) {
+                    closeQuietly(bufferedOut);
```

Comment:
The NiFi commons `FileUtils` method can be used.
```suggestion
                    FileUtils.closeQuietly(bufferedOut);
```